include/zephyr/net/dns_resolve.h

```diff
@@ -82,6 +82,13 @@ enum dns_query_type {
 
 #define DNS_MAX_MCAST_SERVERS (MDNS_SERVER_COUNT + LLMNR_SERVER_COUNT)
 
+/* RFC 1035, 3.1. Name space definitions
+ * To simplify implementations, the total length of a domain name (i.e.,
+ * label octets and label length octets) is restricted to 255 octets or
+ * less.
+ */
+#define DNS_MAX_NAME_LEN	255
```

Comment:
A Kconfig option would surely be a good solution however this was already defined within `dns_resolve.c` thats why I reused it. Do you think this can be replaced in both files with this new kconfig option?

Reply:
Hmm, that is a good question. I am a bit worried that the value might be too large for normal use and users might want to tweak it to save memory. So if we have a Kconfig option it should be used in all over the place.
@rlubos what is you opinion about this?

Reply:
@jukkar I added the Kconfig option for now it would not be a big change to revert this.